src/utils/MembershipUtil.ts

```diff
@@ -0,0 +1,55 @@
+import { CreateMembershipRequest } from "../generated/models/CreateMembershipRequest.generated";
+import { Membership } from "../generated/models/Membership.generated";
+import { OrganizationMembership } from "../generated/models/OrganizationMembership.generated";
+import { Role } from "../generated/models/Role.generated";
+import { User } from "../generated/models/User.generated";
+
+export function generateMembershipFromMembershipRequest(
+	organizationIdentifier: string,
+	organizationMembership: CreateMembershipRequest,
+): Membership {
+	return {
+		organizationIdentifier,
+		role: organizationMembership.role,
+	};
+}
+
+export function generateOrganizationMembership(user: User, role: Role): OrganizationMembership {
+	return {
+		email: user.email,
+		userIdentifier: user.identifier,
+		firstName: user.firstName,
+		lastName: user.lastName,
+		role: role,
+	};
+}
+
+export function generateOrganizationMembershipFor(
+	organizationIdentifier: string,
+	user: User | null,
+): OrganizationMembership | null {
+	if (!user) {
+		return null;
+	}
```

Comment:
Ich finde solche Null Checks an dieser Stelle immer etwas spät und würde diese am liebsten weiter außen in der Call Chain machen, damit nicht so viele Null-Werte herumgereicht werden müssen. Es macht ja theoretisch nie Sinn, diese Funktion ohne einen `User` aufzurufen und daher könnte das schon rausgefiltert/ignoriert werden, bevor diese Funktion jemals ins Spiel kommt.

Soll aber an dieser Stelle nur Inspiration sein und ist definitiv kein Merge Blocker. ✌️

Reply:
Guter Punkt. Bleibt erstmal so, steht bei mir aber auf ToDo.